### DISCUSSION TITLE
How do I enable footnotes?

### DISCUSSION THREAD

**Discussion by jslabovitz** [Omikhleia/markdown.sile#48] [Q&A] [answered]
No matter what I do, I can't seem to get footnotes to parse or be rendered.

Test code:

```
\begin[class=book]{document}

\use[module="packages.markdown"]

\begin[type=markdown]{raw}
Here is a note.[^mynote]

[^mynote]: This is the note's footnote.
\end{raw}

\end{document}
```

I could upload the PDF, but basically it just says:

```
Here is a note.[^mynote]
[^mynote]: This is the note’s footnote.
```

I see there is a 'notes' option (extension?) for Lunamark to deal with footnotes, which appears to default to off, but as far as I can tell from reading the markdown.sile source, this should be turned on.

Help?

**Comment by Omikhleia** [comment, marked as answer]
(Yes, markdown.sile turns the "notes" feature "on" by default.)

It actually seems **lunamark** expects a blank line after the footnote :eyes: 
I just tried the following and I got the footnote as expected:

```
\begin[class=book]{document}
\use[module="packages.markdown"]
\begin[type=markdown]{raw}
Here is a note.[^mynote]

[^mynote]: This is the note's footnote.

\end{raw}
\end{document}
```

Note the blank line before closing the raw environment.

Well, the workaround is simple, but it's clearly a bug or interpretation issue in Lunamark, as I can reproduce it with a standalone Lua script (converting to HTML with only the bare lunamark library):

```
local lunamark = require("lunamark")
local opts = { notes = true }
local writer = lunamark.writer.html5.new(opts)
local parse = lunamark.reader.markdown.new(writer, opts)
local result, matadata = parse([[
Here is a note.[^mynote]

[^mynote]: This is the note's footnote.
]])

print(result)
```

Here again, it fails to parse the note, but adding a blank line makes the parser happy.
For comparison, Pandoc seems happy whether there's a blank line or not --- so you might want to report the issue to **lunamark** :smile: 

(Note that as far as markdown.sile is concerned, it therefore also affects the `\include[src=myfile.md]` construct with an external Markdown file not ending in a blank line.)

**Reply by Omikhleia** [reply]
BTW, resilient.book has much nicer footnotes than the bare SILE book class :lollipop:

**Reply by Omikhleia** [reply]
@jslabovitz I reported to **lunarmark** = see https://github.com/jgm/lunamark/issues/63

I leave it up to you to mark this question as "Answered" if you deem it fit.
